exporter/zipkin/lib/opentelemetry/exporter/zipkin/transformer.rb

```diff
@@ -0,0 +1,192 @@
+              {
+                timestamp: (event.timestamp.to_f * 1_000_000).to_s,
+                value: event.name
+              }
+            else
+              {
+                timestamp: (event.timestamp.to_f * 1_000_000).to_s,
+                value: { "#{event.name}": event.attributes&.transform_values { |val| val.to_s } }.to_json
+              }
+            end
+          end
+
+          zipkin_span[:annotations] = events.map(&:to_h) unless events.empty?
+        end
+
+        def aggregate_span_tags(span_data, tags)
+          # convert attributes to strings
+          # https://github.com/open-telemetry/opentelemetry-specification/blob/84b18b23339dcc0b1a9d48f976a1afd287417602/specification/trace/sdk_exporters/zipkin.md#attribute
+          return tags if span_data[:attributes].nil?
+
+          tags.merge(span_data[:attributes]) do |_key, _oldval, newval|
+            newval.to_s
+          end
+        end
+
+        # mostly based on https://github.com/open-telemetry/opentelemetry-specification/blob/cb16422a61219d4fd99b466a70e47cb8af9e26b1/specification/trace/sdk_exporters/zipkin.md#otlp---zipkin
+        # and https://github.com/open-telemetry/opentelemetry-collector/blob/347cfa9ab21d47240128c58c9bafcc0014bc729d/translator/trace/zipkin/traces_to_zipkinv2.go#L284
+        def endpoint_from_tags(tags, service_name = nil)
+          endpoint = {}
+
+          if service_name
+            endpoint['serviceName'] = service_name
+            endpoint['port'] = tags[ATTRIBUTE_NET_HOST_PORT].to_s if tags[ATTRIBUTE_NET_HOST_PORT]
+
+            if tags[ATTRIBUTE_NET_HOST_IP]
+              ip_addr = IPAddr.new(tags[ATTRIBUTE_NET_HOST_IP])
+
+              # https://github.com/openzipkin/zipkin-api/blob/7692ca7be4dc3be9225db550d60c4d30e6e9ec59/zipkin2-api.yaml#L292
+              # https://github.com/openzipkin/zipkin-api/blob/7692ca7be4dc3be9225db550d60c4d30e6e9ec59/zipkin2-api.yaml#L286
+              if ip_addr.ipv6?
+                endpoint['ipv6'] = ip_addr.to_s
+              else
+                endpoint['ipv4'] = ip_addr.to_s
+              end
+            end
+          else
+            endpoint['port'] = tags[ATTRIBUTE_NET_PEER_PORT].to_s if tags[ATTRIBUTE_NET_PEER_PORT]
+
+            if tags[ATTRIBUTE_NET_PEER_IP]
+              ip_addr = IPAddr.new(tags[ATTRIBUTE_NET_PEER_IP])
+
+              # https://github.com/openzipkin/zipkin-api/blob/7692ca7be4dc3be9225db550d60c4d30e6e9ec59/zipkin2-api.yaml#L292
+              # https://github.com/openzipkin/zipkin-api/blob/7692ca7be4dc3be9225db550d60c4d30e6e9ec59/zipkin2-api.yaml#L286
+              if ip_addr.ipv6?
+                endpoint['ipv6'] = ip_addr.to_s
+              else
+                endpoint['ipv4'] = ip_addr.to_s
+              end
+            end
+          end
+
+          endpoint
+        end
+
+        def int64(byte_string)
+          int = byte_string.unpack1('Q>')
+          int < (1 << 63) ? int : int - (1 << 64)
```

Comment:
Can you just use `q>` here instead of extracting the value as unsigned and converting it to signed?
```suggestion
          byte_string.unpack1('q>')
```

Reply:
I think so, I was just copying the jaeger exporter basically

Reply:
Well then, I suck, don't I? 😆 

Can you check & fix it here, and I'll make the same fix in the Jaeger exporter.

Reply:
Upon closer investigation I think the zipkin receiver expects the same format we use for [b3 propagation](https://github.com/open-telemetry/opentelemetry-ruby/blob/4d576d5a680736e2798872262aae81c8819d2a78/propagator/b3/lib/opentelemetry/propagator/b3/multi/text_map_injector.rb#L38), which makes sense. So long story short, i'm actually no longer sure about whether to use `.unpack1('q>')` in the `jaeger` exporter.